lib/client/tcp-client.ts

```diff
@@ -28,14 +29,77 @@ export class TCPClient {
 
         client.connect(port, host, function () {
             // Write a message to the socket as soon as the client is connected, the server will receive it as message from the client
-            client.write(data);
+            client.write(requestData);
         });
 
 
         // Add a 'data' event handler for the client socket
         // data is what the server sent to this socket
-        client.on("data", function(data: Buffer) {
-            callback(data, null, null);
+        client.on("data", function(incommingData: Buffer) {
```

Comment:
Almost all the handling done here is actually in the httpBuffer class, which the data coming back is pushed into on the callback:
https://github.com/bespoken/bst/blob/7701d87e3dc07554f4c7d2c1123667a31ec7d90f/lib/client/bespoke-client.ts#L106

In that class, I pull out the response code, method, path, headers, and other data from the payload automatically.

For a 302, why do we need to do additional processing? What is the exact payload that comes back, and why does it not work?

Reply:
On a 302 , we just get the header with the new Location, so if we don't resend the data to that one, we end loading forever
Ex.
bst  proxy http --bstHost localhost --bstPort 5000 --targetDomain bit.ly 80

Reply:
The targetDomain should not change. The reply should do the redirect though.

I believe the problem is here:
https://github.com/bespoken/spokes/blob/master/lib/server/HTTPPipe.ts#L61

We are not setting the statusCode or statusMessage on the sourceResponse:
https://nodejs.org/api/http.html#http_response_statusmessage
https://nodejs.org/api/http.html#http_response_statuscode

Both of these should be set from the values on the response.

Reply:
I've also attached a simple project to test with:
[redirect-test.zip](https://github.com/bespoken/bst/files/1455502/redirect-test.zip)